module/zfs/metaslab.c

```diff
@@ -236,12 +265,81 @@ metaslab_class_destroy(metaslab_class_t *mc)
 	ASSERT(mc->mc_deferred == 0);
 	ASSERT(mc->mc_space == 0);
 	ASSERT(mc->mc_dspace == 0);
+#ifdef ZFS_DEBUG
+	for (int t = 0; t < TXG_SIZE; t++)
+		ASSERT0(mc->mc_alloc_stats_dirty[t]);
+#endif
 
+	avl_destroy(&mc->mc_alloc_tree);
 	refcount_destroy(&mc->mc_alloc_slots);
 	mutex_destroy(&mc->mc_lock);
+	mutex_destroy(&mc->mc_alloc_lock);
 	kmem_free(mc, sizeof (metaslab_class_t));
 }
 
+void
+metaslab_class_enable_stats(metaslab_class_t *mc, int64_t class_stats_obj)
```

Comment:
`enable` implies to me that this is called at most once per pool (in the whole history of the pool).  Is that right, or is this actually just `load`ing the stats from the object into memory?

Reply:
it is a load; I'll rename it accordingly